tests/model_options/test_default_related_name.py

```diff
@@ -18,6 +21,20 @@ def test_no_default_related_name(self):
     def test_default_related_name(self):
         self.assertEqual(list(self.author.books.all()), [self.book])
 
+    def test_default_related_name_in_queryset_lookup(self):
+        self.assertEqual(Author.objects.get(books=self.book), self.author)
+
+    def test_show_deprecated_message_when_model_name_in_queryset_lookup(self):
+        msg = ("Query lookup '%s' is deprecated in favor of"
+               " Meta.default_related_name '%s'."
+               % ('book', 'books'))
+        with warnings.catch_warnings(record=True) as warns:
+            warnings.simplefilter('always')
+            Author.objects.get(book=self.book)
+        warning = warns.pop()
```

Comment:
Use `warnings.simplefilter('once')` in this case.

There has been a lot of stuff moving around lately in the `Field` and `_meta` API andI just want to make sure the backward compatibility shim you added doesn't use deprecated stuff itself.